devtools-integration-tests/src/test/java/org/ballerina/devtools/cmd/BuildNewCommandTest.java

```diff
@@ -0,0 +1,165 @@
+    private ByteArrayOutputStream console;
+    protected PrintStream printStream;
+
+    private static final String DISTRIBUTION_FILE_NAME = "ballerina-" + MAVEN_VERSION;
+
+    @BeforeClass()
+    public void setUp() throws IOException, InterruptedException {
+        setupDistributions();
+
+        tempHomeDirectory = Files.createTempDirectory("bal-test-integration-packaging-home-");
+        tempWorkspaceDirectory = Files.createTempDirectory("bal-test-integration-packaging-workspace-");
+        createSettingToml(tempHomeDirectory);
+        envVariables = addEnvVariables(getEnvVariables());
+        this.console = new ByteArrayOutputStream();
+        this.printStream = new PrintStream(this.console);
+    }
+
+    @BeforeMethod
+    public void beforeMethod() {
+        this.console = new ByteArrayOutputStream();
+        this.printStream = new PrintStream(this.console);
+    }
+
+    protected String readOutput() throws IOException {
+        return readOutput(false);
+    }
+
+    protected String readOutput(boolean silent) throws IOException {
+        String output = "";
+        output = console.toString();
+        console.close();
+        console = new ByteArrayOutputStream();
+        printStream = new PrintStream(console);
+        if (!silent) {
+            PrintStream out = System.out;
+            out.println(output);
+        }
+        return output;
+    }
+
+    @Test(description = "Build package created from new command with default template")
+    public void testCompilingNewCommandDefaultTempProject() throws IOException, InterruptedException {
+        Process newCommand = executeNewCommand(DISTRIBUTION_FILE_NAME, this.tempWorkspaceDirectory,
+                new LinkedList<>(Collections.singletonList("project_name")), this.envVariables);
+
+        Path packageDir = this.tempWorkspaceDirectory.resolve("project_name");
+        Assert.assertTrue(Files.exists(packageDir));
+
+        Process build = executeBuildCommand(DISTRIBUTION_FILE_NAME, this.tempWorkspaceDirectory,
+                new LinkedList<>(Collections.singletonList("project_name")), this.envVariables);
+
+        Assert.assertTrue(packageDir.resolve("target").resolve("bin").resolve("project_name.jar").toFile().exists());
+    }
+
+    @Test(description = "Build package created from new command with main template")
+    public void testCompilingNewCommandMainTempProject() throws IOException, InterruptedException {
+        Process newCommand = executeNewCommand(DISTRIBUTION_FILE_NAME, this.tempWorkspaceDirectory,
+                new LinkedList<String>(Arrays.asList("main_sample", "-t", "main")), this.envVariables);
+
+        Path packageDir = this.tempWorkspaceDirectory.resolve("main_sample");
+        Assert.assertTrue(Files.exists(packageDir));
+
+        Process build = executeBuildCommand(DISTRIBUTION_FILE_NAME, this.tempWorkspaceDirectory,
+                new LinkedList<>(Collections.singletonList("main_sample")), this.envVariables);
```

Comment:
For the `executeBuildCommand()` function, the parameters `DISTRIBUTION_FILE_NAME`, `tempWorkspaceDirectory` and `this.envVariables` are repeating. Same for the `executeNewCommand()`

Would it make sense to resolve these in the utils class itself? ie. Move initialization of these variables to the util class and have the function only accept the LinkedList<>. Are the values changing for each test?